### PR TITLE
Issue 1462 - editable pins

### LINKED ISSUE #1462: Editable pins for risks and issues
### Description
Pins should be able to be editable

### Goals
- [ ] The user can edit an issue, change the pin and  then save it
- [ ] The user can edit a risk, change the pin and  then save it

### Tasks
- [x] Update backend to be able to change the pin for an issue
- [x] Update backend to be able to change the pin for a risk
- [ ] Update frontend to be able to change the pin for an issue
- [ ] Update frontend to be able to change the pin for a risk

### PR DESCRIPTION
**Pull request by sanmont3drepo**
This fixes #1462

### CODE REVIEW

File: frontend/routes/viewer/components/issues/components/issueDetails/issueDetails.component.tsx [hunk 1]
```diff
@@ -210,7 +219,11 @@ export class IssueDetails extends React.PureComponent<IProps, IState> {
 				onExpandChange={this.handleExpandChange}
 				renderCollapsable={this.renderDetailsForm}
 				renderNotCollapsable={() => this.renderLogList(comments && !!comments.length && !this.isNewIssue)}
-				handleHeaderClick={() => this.setCameraOnViewpoint({viewpoint: this.issueData.viewpoint})}
+				handleHeaderClick={() => {
+					if (this.issueData._id) { // if its a new issue it shouldnt go to the viewpoint
```
Comment (simplypixi) [inline frontend/routes/viewer/components/issues/components/issueDetails/issueDetails.component.tsx]:
```suggestion
					if (!this.isNewIssue) {
```
And then you can remove the comment

File: frontend/routes/viewer/components/issues/components/issueDetails/issueDetails.component.tsx [hunk 2]
```diff
@@ -288,6 +299,15 @@ export class IssueDetails extends React.PureComponent<IProps, IState> {
 		}
 	}
 
+	public onPositionSave = (position) => {
+		if (this.props.issue._id) {
```
Comment (simplypixi) [inline frontend/routes/viewer/components/issues/components/issueDetails/issueDetails.component.tsx]:
```suggestion
		if (!this.isNewIssue) {
```

File: frontend/routes/viewer/components/pinButton/pinButton.component.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+	public render() {
+		const wasPinDropped = this.state.wasPinDropped || this.props.hasPin;
+		const editMsg = !wasPinDropped ? 'Add pin' : 'Edit pin';
+		const pinLabel =  this.state.active ? 'Save pin' :  editMsg;
+
+		return (
+				<Container>
+					<PinIcon color={this.state.active ? 'secondary' : 'primary'}/>
+					<LabelButton onClick={this.onClickButton}>{pinLabel}</LabelButton>
+				</Container>);
```
Comment (simplypixi) [inline frontend/routes/viewer/components/pinButton/pinButton.component.tsx]: the bracket should be placed in next line

File: frontend/routes/viewer/components/pinButton/pinButton.styles.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+`;
+
+export const LabelButton = styled(Button)`
+	&& {
+		width: 76px;
+		text-transform: none;
+		padding: 0;
+		min-height: 26px;
+		border-radius: 28px;
+		color: #fafafa;
```
Comment (simplypixi) [inline frontend/routes/viewer/components/pinButton/pinButton.styles.ts]:
```suggestion
		color: ${COLOR.LIGHT_GRAY};
```

File: frontend/routes/viewer/components/risks/components/riskDetails/riskDetails.component.tsx [hunk 5]
```diff
@@ -146,6 +147,8 @@ export class RiskDetails extends React.PureComponent<IProps, IState> {
 		}
 	}
 
+	// onSavePin: (position) => void;
```
Comment (simplypixi) [inline frontend/routes/viewer/components/risks/components/riskDetails/riskDetails.component.tsx]: Is it necessary?

File: frontend/routes/viewer/components/risks/components/riskDetails/riskDetails.component.tsx [hunk 6]
```diff
@@ -156,7 +159,11 @@ export class RiskDetails extends React.PureComponent<IProps, IState> {
 				permissions={this.props.modelSettings.permissions}
 				currentUser={this.props.currentUser}
 				myJob={this.props.myJob}
-			/>
+				onChangePin={this.handleChangePin}
+				onSavePin={this.onPositionSave}
+				pinId={this.riskData._id}
+				hasPin={this.riskData.position && this.riskData.position.length}
+				/>
```
Comment (simplypixi) [inline frontend/routes/viewer/components/risks/components/riskDetails/riskDetails.component.tsx]: wrong indentation

File: frontend/routes/viewer/components/risks/components/riskDetails/riskDetailsForm.component.tsx [hunk 7]
```diff
@@ -249,6 +254,13 @@ class RiskDetailsFormComponent extends React.PureComponent<IProps, IState> {
 								/>
 							)} />
 						</StyledFormControl>
+						<StyledFormControl>
+							<PinButton onChange={this.props.onChangePin}
+									onSave={this.props.onSavePin}
+									pinId={this.props.pinId}
+									hasPin={this.props.hasPin}
+									/>
```
Comment (simplypixi) [inline frontend/routes/viewer/components/risks/components/riskDetails/riskDetailsForm.component.tsx]: wrong indentation

File: frontend/routes/viewer/components/pinButton/pinButton.component.tsx [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import * as React from 'react';
+import { PinIcon, LabelButton, Container } from './pinButton.styles';
```
Comment (simplypixi) [inline frontend/routes/viewer/components/pinButton/pinButton.component.tsx]: Try to keep the proper order of imports - the closest paths at the bottom.

File: frontend/routes/viewer/components/pinButton/pinButton.component.tsx (hunk 3 above)
Comment (sanmont3drepo) [reply frontend/routes/viewer/components/pinButton/pinButton.component.tsx]: Which bracket? Or you mean the parenthesis?

File: frontend/routes/viewer/components/pinButton/pinButton.component.tsx (hunk 8 above)
Comment (sanmont3drepo) [reply frontend/routes/viewer/components/pinButton/pinButton.component.tsx]: Sorry, the imports are generated automatically by Visual Studio Code. 
Can you provide what would be the most desired order in this case?

Comment (simplypixi) [reply frontend/routes/viewer/components/pinButton/pinButton.component.tsx]:
```suggestion
import * as React from 'react';
import { Viewer } from '../../../../services/viewer/viewer';
import { VIEWER_EVENTS, NEW_PIN_ID } from '../../../../constants/viewer';
import { PIN_COLORS } from '../../../../styles';
import { PinIcon, LabelButton, Container } from './pinButton.styles';
```

File: frontend/routes/viewer/components/pinButton/pinButton.component.tsx (hunk 3 above)
Comment (simplypixi) [reply frontend/routes/viewer/components/pinButton/pinButton.component.tsx]: Yes, I meant parenthesis. Sorry about that xd

**Pull request merged**